test/FsAutoComplete.Tests.Lsp/CoreTests.fs

```diff
@@ -916,27 +916,47 @@ let tooltipTests toolsPath =
 
 let highlightingTests toolsPath =
   let serverStart = lazy (
-    let path = Path.Combine(__SOURCE_DIRECTORY__, "TestCases", "CodeLensTest")
+    let path = Path.Combine(__SOURCE_DIRECTORY__, "TestCases", "HighlightingTest")
     let (server, event) = serverInitialize path defaultConfigDto toolsPath
-    let projectPath = Path.Combine(path, "CodeLensTest.fsproj")
-    parseProject projectPath server |> Async.RunSynchronously
-    let path = Path.Combine(path, "Script.fs")
+    let path = Path.Combine(path, "Script.fsx")
     let tdop : DidOpenTextDocumentParams = { TextDocument = loadDocument path}
 
     do server.TextDocumentDidOpen tdop |> Async.RunSynchronously
-    (server, path)
-    )
-  let serverTest f () =
-    let (server, path) = serverStart.Value
-    f server path
+    match waitForParseResultsForFile "Script.fsx" event with
+    | Ok () -> ()
+    | Error e -> failwithf "Errors while parsing highlighting script %A" e
 
-  testCase "Document Highlighting" (serverTest (fun server path ->
-    let p : HighlightingRequest = { FileName =  path}
+    let p : HighlightingRequest = { FileName =  path }
     let res = server.GetHighlighting p |> Async.RunSynchronously
-    printfn "%A" res
-    ()
-    // Expect.equal res.Length 2 "Document Symbol has all symbols"
-  ))
+    res
+  )
+
+  let serverTest f () =
+    let highlights = serverStart.Value
+    f highlights
+
+  testList "Document Highlighting Tests" [
+    testCase "can get highlights" (serverTest (fun highlights ->
+      match highlights with
+      | Ok _ ->
+        ()
+      | Error e ->
+        failwithf "error of %A" e
+    ))
+    testCase "can get inner ranges" (serverTest (fun highlights ->
+      match highlights with
+      | Ok highlights ->
+        // in the middle of the `PeePee` usage in the parameter of `yeet`
+        let troublesomeRange (parentRange: FSharp.Compiler.Range.range) = FSharp.Compiler.Range.mkFileIndexRange parentRange.FileIndex (FSharp.Compiler.Range.mkPos 7 21) (FSharp.Compiler.Range.mkPos 7 21)
```

Comment:
could definitely extract this and ask a number of different questions about the same set of returned highlights...

Reply:
I extracted this and added a couple more tests that I'll want to ensure give the correct results.